commcare_connect/opportunity/forms.py

```diff
@@ -1134,6 +1136,8 @@ def __init__(self, *args, **kwargs):
 
         self.fields["invoice_number"].initial = self.generate_invoice_number()
         self.fields["date"].initial = str(datetime.date.today())
+        self.fields["start_date"].initial = str(self.get_earliest_uninvoiced_date())
```

Comment:
nit: I think we want to default to the previous month (we expect invoices to be monthly) for start and end dates, rather than over all time and until today. I am more certain we want the end date to be the end of the past month, but could be convinced on the start date

Reply:
In the design doc (1) it states:
1. Start date: "...Defaults to the date in the system on which the first approved, but un-invoiced visit"
2. End date: "Last day on which a given visit may have been approved to be included on this invoice; defaults to today - 1". (actually, I need to update the code to not include today)

I prefer the spec's definitions because there's a UX component baked in that I think is very helpful, i.e. the user can't miss uninvoiced visits accidentally.

With the current behaviour, as invoices gets generated on a monthly basis this defaulting to the previous month (as you envision) will automatically happen, but as a side effect and not a design choice.